### PR TITLE
VACMS-13222: Display new 'Where' format for events

### PR DESCRIPTION
**Pull request by chriskim2311**
## Description
For this ticket we are adding a new address format along with google maps link to direction. To accomplish this, we have added an update to the graphQL query to add the fieldAddress to the fieldFacilityLocation to pull the address into the event data obj. And we made the necessary changes on the FE drupal templates for events.

## Testing done & Screenshots

Testing URLs:
/outreach-and-events/events/60333/
/outreach-and-events/events/60524/

Before:
Outreach_And_Events___Women_s_Resource_Fair_With_Veterans_Justice_Outreach_Program_-_Los_Angeles__CA___Veterans_Affairs_before

After:
Outreach_And_Events___Women_s_Resource_Fair_With_Veterans_Justice_Outreach_Program_-_Los_Angeles__CA___Veterans_Affairs

## QA steps

What needs to be checked to prove this works?  
Verify that event pages have new address format and link to google maps. 
Verify that VA and non VA facility events both show the new formatted address. 

What needs to be checked to prove it didn't break any related things?  
Verify that event pages are still working properly.

## Acceptance criteria
- [x] Address is presented as described above, including Directions link 
- [x] Optional building/floor/room text is displayed if provided by CMS
- [x] Google maps directions link is present / working for all addresses
- [x] VA facilities additionally display the name of the VA facility
  - [x] If the VA facility has a web page (eg VAMC), the facility name is hyperlinked to that page (see Eng Notes above)
  - [x] Otherwise, there is no hyperlink (eg VC Outstation)
- [ ] Design review required
- [ ] A11y review required

### CODE REVIEW

**Comment by randimays** [comment]
Quick tip: Please make sure there are URLs for helping others test in the review instance

File: src/site/stages/build/drupal/graphql/nodeEvent.graphql.js [hunk 1]
```diff
@@ -68,9 +68,58 @@ const nodeEvent = `
     fieldEventRegistrationrequired
     fieldFacilityLocation {
       entity {
+        ... on NodeVetCenter {
```
Comment (randimays) [inline src/site/stages/build/drupal/graphql/nodeEvent.graphql.js:71]: Looks like all of the facility types pull the same `fieldAddress` properties. Maybe we could try [this strategy](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/site/stages/build/drupal/graphql/alerts.graphql.js) where we define a constant at the top and then use that variable for each of these?

**Review by randimays — COMMENTED** [review]
Just a reminder to double-check that this looks good on various screen widths

File: src/site/stages/build/drupal/graphql/nodeEvent.graphql.js (hunk 1 above)
Comment (chriskim2311) [reply src/site/stages/build/drupal/graphql/nodeEvent.graphql.js:80]: Good call made this more DRY

File: src/site/layouts/event.drupal.liquid [hunk 2]
```diff
@@ -62,20 +62,45 @@
...
+                <p class="vads-u-margin--0">
+                  {% if fieldFacilityAddress.locality != empty  %}
+                    {{ fieldFacilityAddress.locality | strip }},
+                  {% endif %}
+
+                  {% if fieldFacilityAddress.administrativeArea != empty  %}
+                    {{ fieldFacilityAddress.administrativeArea }}
+                  {% endif %}
+                  </p>
+                    {% capture fullAddress %}
```
[the remark this answers is not in the document]
Comment (chriskim2311) [reply src/site/layouts/event.drupal.liquid:96]: Thanks, that makes sense. I have made this change to check for the addressLine1

**Comment by chriskim2311** [comment]
@maxx1128 @randimays Thanks for the comments made the changes ready for review!

File: src/site/layouts/event.drupal.liquid [hunk 3]
```diff
@@ -62,20 +62,47 @@
...
+                {% assign fieldFacilityAddress = fieldFacilityLocation.entity.fieldAddress  %}
                   <div class="vads-u-display--flex vads-u-flex-direction--column">
                     <p class="vads-u-margin--0">
                       <a href="{{ fieldFacilityLocation.entity.entityUrl.path }}">
                         {{ fieldFacilityLocation.entity.title }}
                       </a>
                     </p>
-
                     <p class="vads-u-margin--0">{{ fieldLocationHumanreadable }}</p>
+                      {% if fieldFacilityAddress.addressLine1 != empty %}
```
Comment (maxx1128) [inline src/site/layouts/event.drupal.liquid:79]: The indentation is still out of whack here, but is not a blocker at this point. It should be more like this so the contents of the conditionals are indented further than them.

```liquid
                    {% if fieldFacilityAddress.addressLine1 != empty %}
                      <p class="vads-u-margin--0">{{ fieldFacilityAddress.addressLine1  }}</p>
                    {% endif %}
                    {% if fieldFacilityAddress.addressLine2 != empty %}
                        <p class="vads-u-margin--0">{{ fieldFacilityAddress.addressLine2 }}</p>
                    {% endif %}
```

**Review by maxx1128 — APPROVED** [review]
Built it locally and the functionality works for me 👍 One small nitpick about the indentation still but not a blocker.

**Review by randimays — APPROVED** [review]
LGTM

**Comment by laflannery** [comment]
@chriskim2311 Just confirming that the directions link will open in a new tab but otherwise this looks good to me!

**Comment by thejordanwood** [comment]
@chriskim2311 This page looks good and the design is approved by me!

File: src/site/layouts/event.drupal.liquid (hunk 3 above)
Comment (chriskim2311) [reply src/site/layouts/event.drupal.liquid]: @maxx1128 Was having issues with the linter. I was able to find the issue in the code preventing the formatting. This file is good to go now thanks

**Pull request merged**